contributing/development/core_and_modules/internal_rendering_architecture.rst

```diff
@@ -207,18 +207,23 @@ for more information.
 Metal
 ^^^^^
 
-Godot supports Metal rendering via `MoltenVK <https://github.com/KhronosGroup/MoltenVK>`__,
-as macOS and iOS do not support Vulkan natively.
-This is done automatically when specifying the Vulkan driver in the Project Settings.
+Godot provides a native Metal driver that works on all Apple Silicon hardware
+(macOS ARM). Compared to using the MoltenVK translation layer, this is
+significantly faster, particularly in CPU-bound scenarios.
 
-MoltenVK makes driver maintenance easy at the cost of some performance overhead.
-Also, MoltenVK has several limitations that a native Metal driver implementation
-wouldn't have. Both the clustered and mobile
-:ref:`doc_internal_rendering_architecture_methods` can be used with a Metal
-backend via MoltenVK.
+Both the Forward+ and Mobile :ref:`doc_internal_rendering_architecture_methods` can be
+used with Metal.
+
+:ref:`doc_internal_rendering_architecture_core_shaders` are shared with the
+Vulkan renderer. Shaders are transpiled from GLSL to :abbr:`MSL (Metal Shading Language)`
+using SPIRV-Cross.
```

Comment:
@stuartcarnie Is this paragraph correct? Does Metal have something akin to SPIR-V or DXIL that would be used as a transpiling target?

Reply:
Yes, that paragraph is correct. There is not a documented intermediate representation, however, shaders can be compiled to an IR, which is compiled on the hardware when used. We would have to use the Metal compiler tools to generate the IR from Metal Shader Language.